apps/settings/js/simcard_fdn.js

```diff
@@ -200,9 +202,18 @@ define('simcard_fdn', ['modules/dialog_service'], function(DialogService) {
         number: number
       });
 
+      var resetHref = function() {
+        // we need this timeout
+        setTimeout(function() {
+          self.gaiaHeader.dataset.href = '#call-fdnSettings';
+        }, 100);
```

Comment:
In 3.0, we don't need this hack because dialogs have no relationship with panels, so I think for v2.2, it's okay to use a small workaround here :)